src/compile/mark/encode/tooltip.ts

```diff
@@ -164,3 +165,23 @@ export function tooltipRefForEncoding(
   const keyValues = entries(data).map(([key, value]) => `"${key}": ${value}`);
   return keyValues.length > 0 ? {signal: `{${keyValues.join(', ')}}`} : undefined;
 }
+
+/**
+ * Transforms a tooltip value that is an array to a string with line breaks
+ */
+function addLineBreaksToTooltip(
+  channelDef: Encoding<string>['text' | 'tooltip'],
+  config: Config,
+  expr: 'datum' | 'datum.datum' = 'datum',
+): VgValueRef {
+  // tooltip fields with a format property are no strings
+  const fieldDefWithFormat = channelDef as {field: string; type: string; format: string};
+  if (fieldDefWithFormat?.type === 'nominal' && !fieldDefWithFormat.format) {
+    const fieldString = `datum["${fieldDefWithFormat.field}"]`;
+    return {
+      signal: `isValid(${fieldString}) ? isArray(${fieldString}) ? join(${fieldString}, '\\n') : ${fieldString} : ""+${fieldString}`,
```

Comment:
Why do we need to convert invalid values to strings? Can we reuse logic here?

Reply:
This is in line with the way [formatSignalRef](https://github.com/vega/vega-lite/blob/f1052b6cadf57e866a9ab1054e2d7767d1258d7a/src/compile/format.ts#L132) handles nominal fields:

```return {signal: `isValid(${field}) ? ${field} : ""+${field}`};```

I think to render `null`, `undefined`, or `NaN` as strings in tooltips and area descriptions?